### PR TITLE
the calico-apiserver namespace frequently get stuck for a long time in terminating status when upgrading to enterprise

### PR DESCRIPTION
**Pull request by gcosgrave**
…
## Description

The apiserver TigeraStatus error message `Error creating / updating resource: secrets "tigera-apiserver-certs" is forbidden: unable to create new content in namespace calico-apiserver because it is being terminated`  identifies the problem as the certs secret is called "calico-apiserver-certs" when it is in the "calico-apiserver" namespace, and "tigera-apiserver-certs" when in the "tigera-apiserver" namespace

```
apiVersion: operator.tigera.io/v1
kind: TigeraStatus
metadata:
    creationTimestamp: "2024-07-18T11:15:59Z"
    generation: 1
    name: apiserver
    resourceVersion: "2010"
    uid: fcdb9846-3772-4684-afe3-db458486608b
spec: {}
status:
    conditions:
        - lastTransitionTime: "2024-07-18T11:18:06Z"
          message: 'Error creating / updating resource: secrets "tigera-apiserver-certs" is forbidden: unable to create new content in namespace calico-apiserver because it is being terminated'
          observedGeneration: 1
          reason: ResourceUpdateError
          status: "True"
          type: Degraded
        - lastTransitionTime: "2024-07-18T11:17:56Z"
          observedGeneration: 1
          reason: Unknown
          status: "False"
          type: Available
        - lastTransitionTime: "2024-07-18T11:17:56Z"
          message: Deployment "tigera-system/tigera-apiserver" is not available (awaiting 2 replicas)
          observedGeneration: 1
          reason: ResourceNotReady
          status: "True"
          type: Progressing
```

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/apiserver/apiserver_controller.go [hunk 1]
```diff
@@ -252,11 +252,7 @@ func (r *ReconcileAPIServer) Reconcile(ctx context.Context, request reconcile.Re
 		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying installation", err, reqLogger)
 		return reconcile.Result{}, err
 	}
-	if variant == "" {
```
Comment (caseydavenport) [inline pkg/controller/apiserver/apiserver_controller.go:255]: @tmjd @gcosgrave I think we still do need this check to handle the tiniest of race conditions.

If a user creates an Installation with no variant, we want to wait for the Installation controller to default it before we continue. It should only add a second of delay.

Comment (tmjd) [reply pkg/controller/apiserver/apiserver_controller.go:255]: Yeah I agree. To be specific the check can be changed to `if installationSpec.Variant == ""`

Comment (gcosgrave) [reply pkg/controller/apiserver/apiserver_controller.go:255]: done

File: pkg/controller/apiserver/apiserver_controller.go [hunk 2]
```diff
@@ -252,11 +252,11 @@ func (r *ReconcileAPIServer) Reconcile(ctx context.Context, request reconcile.Re
...
+	if installationSpec.Variant == "" {
 		r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for Installation to be ready", nil, reqLogger)
```
Comment (tmjd) [inline pkg/controller/apiserver/apiserver_controller.go:256]: I think the message here needs to be changed to something like "Waiting for Installation Variant to be set". I think the ResourceNotReady is good though.

**Review by tmjd — APPROVED** [review]
Looks good, I think the unset variant message could use an update.

File: pkg/controller/apiserver/apiserver_controller.go (hunk 2 above)
Comment (gcosgrave) [reply pkg/controller/apiserver/apiserver_controller.go]: Agreed, message updated

**Pull request merged**